src/test/java/com/hazelcast/aws/impl/DescribeInstancesTest.java

```diff
@@ -235,11 +241,46 @@ public void test_DescribeInstances_when_Only_TagValue_Set()
             throws Exception {
         AwsConfig awsConfig = new AwsConfig();
         awsConfig.setEnabled(true).setAccessKey(System.getenv("AWS_ACCESS_KEY_ID"))
-                 .setSecretKey(System.getenv("AWS_SECRET_ACCESS_KEY")).setTagValue("aws-tag-value-1");
+                .setSecretKey(System.getenv("AWS_SECRET_ACCESS_KEY")).setTagValue("aws-tag-value-1");
 
         getInstancesAndVerify(awsConfig);
     }
 
+    @Test(timeout = TIMEOUT_FACTOR * CALL_SERVICE_TIMEOUT)
+    public void test_CallService_Timeout() {
+        final String nonRoutable = "10.255.255.254";
+        AwsConfig awsConfig = new AwsConfig();
+        awsConfig.setConnectionTimeoutSeconds((int) TimeUnit.MILLISECONDS.toSeconds(CALL_SERVICE_TIMEOUT));
+        DescribeInstances describeInstances = new DescribeInstances(awsConfig);
+
+        try {
+            describeInstances.callService(nonRoutable);
+        } catch (Exception e) {
+            Assert.assertTrue(e instanceof SocketTimeoutException);
+            return;
+        }
+
+        Assert.fail("Timeout setting failure in call service operation");
+    }
+
+    @Test(timeout = TIMEOUT_FACTOR * CALL_SERVICE_TIMEOUT)
+    public void test_RetrieveMetaData_Timeout() {
+        final String nonRoutable = "http://10.255.255.254";
+        AwsConfig awsConfig = new AwsConfig();
+        awsConfig.setConnectionTimeoutSeconds((int) TimeUnit.MILLISECONDS.toSeconds(CALL_SERVICE_TIMEOUT));
+
+        DescribeInstances describeInstances = new DescribeInstances(awsConfig);
+
+        try {
+            describeInstances.retrieveRoleFromURI(nonRoutable);
+        } catch (Exception e) {
+            Assert.assertTrue(e instanceof InvalidConfigurationException);
+            return;
+        }
+
+        Assert.fail("Timeout setting failure in meta data retrieval operation");
+    }
```

Comment:
What about using `expected` here and above new tests?
@Test(expected = InvalidConfigurationException.class)

Reply:
👍 updating PR accordingly.